include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h

```diff
@@ -152,18 +152,58 @@ __pattern_transform_scan_base_impl(_ExecutionPolicy&& __exec, _Iterator1 __first
     }
 }
 
+// TODO In C++20 we may try to use std::equality_comparable
+template <typename _Iterator1, typename _Iterator2, typename = void>
+struct __is_equality_comparable : std::false_type
+{
+};
+
+// All with implemented operator ==
+template <typename _Iterator1, typename _Iterator2>
+struct __is_equality_comparable<_Iterator1, _Iterator2,
+                                std::void_t<decltype(::std::decay_t<_Iterator1>() == ::std::decay_t<_Iterator2>())>>
```

Comment:
```suggestion
                                std::void_t<decltype(::std::decay_t<_Iterator1>{} == ::std::decay_t<_Iterator2>{})>>
```

Ideally, we should use `declval` here, but I think it is safe to expect iterator types being default-constructible.